src/NuGet.Clients/NuGet.CommandLine/Commands/ProjectFactory.cs

```diff
@@ -143,24 +143,15 @@ private void Initialize(dynamic project)
             }
 
             // This happens before we obtain warning properties, so this Logger is still IConsole.
-            IConsole console = Logger as IConsole;
-            switch (LogLevel)
+            if (Logger is IConsole console)
```

Comment:
A lot of tests were failing because Logger was `NullLogger` which resulted in an invalid cast exception.  Instead of `as` we could use `is` and do a onetime check if the Logger is an `IConsole` type.

Reply:
Another thing we could do is 
``` csharp
if (LogLevel != LogLevel.Debug)
{
    var console = (IConsole)Logger;
    console.Verbosity = LogLevel switch
    {
        LogLevel.Verbose => Verbosity.Detailed,
        LogLevel.Information => Verbosity.Normal,
        LogLevel.Minimal => Verbosity.Quiet,
        _ => console.Verbosity
    };
}
```
And in our tests make sure `LogLevel == LogLevel.Debug`